iot_config/src/route_service.rs

```diff
@@ -135,6 +137,36 @@ impl RouteService {
 
         DevAddrEuiValidator::new(route_id, admin_keys, &self.pool, check_constraints).await
     }
+
+    async fn validate_skf_devaddrs<'a>(
+        &self,
+        route_id: &'a str,
+        updates: &[route_skf_update_req_v1::RouteSkfUpdateV1],
+    ) -> Result<(), Status> {
+        let ranges: Vec<DevAddrRange> = route::list_devaddr_ranges_for_route(route_id, &self.pool)
+            .map_err(|err| match err {
+                RouteStorageError::UuidParse(_) => {
+                    Status::invalid_argument(format!("unable to parse route_id: {route_id}"))
+                }
+                _ => Status::internal("error retrieving devaddrs for route"),
+            })?
+            .filter_map(|range| async move { range.ok() })
+            .collect()
+            .await;
+
+        'update_addrs: for update in updates {
+            for range in &ranges {
```

Comment:
You can avoid the `continue` by using `any` on an iterator rather than the inside for loop.